File: src/main/java/com/GoldenMine/parser/postposition/IPostPosition서술어.java
package com.GoldenMine.parser.postposition;

import java.util.List;
import kr.co.shineware.nlp.komoran.constant.DEFAULT_MODEL;
import kr.co.shineware.nlp.komoran.core.Komoran;
import kr.co.shineware.nlp.komoran.model.KomoranResult;
import kr.co.shineware.util.common.model.Pair;

public class IPostPosition서술어 implements IPostPosition {
    Komoran komoran = new Komoran(DEFAULT_MODEL.FULL);
    {
        komoran.setFWDic("user_data/fwd.user");
        komoran.setUserDic("user_data/dic.user");
    }

    @Override
    public String getType() {
        return "서술어";
    }

    @Override
    public int verify(CharSequence lastSeq) {
        String lastToStr = lastSeq.toString();

        KomoranResult analyzeResultList = komoran.analyze(lastToStr);
        List<Pair<String, String>> list = analyzeResultList.getList();
        //Pair<String, String> first = list.get(0);
        Pair<String, String> last = list.get(list.size() - 1);

        switch (last.getSecond()) {
            case "EC":
            case "ETM":
            case "ETN":
            case "JX":
                return 0;
            default:
                return -1;
        }
        //같다면
        //더한
        //뺀
        //3이라면



        //return 0;
    }

    @Override
    public JosaCommunity getJosaCommunity() {
        return JosaCommunity.격조사;
    }
}

File: src/main/java/kr/goldenmine/parser/postposition/IPostPosition서술어.kt
package kr.goldenmine.parser.postposition

import kr.goldenmine.parser.ParseContext

class IPostPosition서술어 : IPostPosition {
    val defaultWord = "TEST"

    override val type: String
        get() = "서술어"

    override val josaCommunity: JosaCommunity
        get() = JosaCommunity.격조사

    override fun verify(lastSeq: CharSequence): Int {
        val builder = StringBuilder()

        var mode = 0

        for(ch in lastSeq) {
            if(ch == '\'') {
                if(mode == 0) {
                    mode = 1
                } else if(mode == 1) {
                    mode = 0
                    builder.append(defaultWord)
                }
                continue
            }
            if(ch == '\"') {
                if(mode == 0) {
                    mode = 2
                } else if(mode == 2) {
                    mode = 0
                    builder.append(defaultWord)
                }
                continue
            }

            if(mode == 0)
                builder.append(ch)
        }

        val lastToStr = builder.toString()

        //println("lasttostr서술어: $lastToStr")



        val analyzeResultList = ParseContext.komoran.analyze(lastToStr)
        val list = analyzeResultList.list
        //Pair<String, String> first = list.get(0);
        val last = list[list.size - 1]

        var verified = false

        when (last.second) {
            "EC", "ETM", "ETN", "JX" -> when (last.first) {
                "까지" -> {}
                else -> verified = true
            }
            else -> when (last.first) {
                "라면" -> verified = true
            }
        }


        if(verified) {
            //println(list)
            for (value in list) {
                when (value.second) {
                    "VV", "VA", "XSV" -> return 0
                }
            }
        }
        //같다면
        //더한
        //뺀
        //3이라면


        //return 0;

        return -1
    }
}
